src/pages/api/UpdateFormData.ts

```diff
@@ -0,0 +1,208 @@
+
+  return combinedDate;
+}
+
+async function UpdateFormDataToBackend(formData: PatchTypes) {
+  const { student, test, timeline, dateCreated, session } = formData;
+
+  let start_time = await formatDateTime(
+    timeline.startDate as string,
+    timeline.startTime as string
+  );
+  let end_time = await formatDateTime(
+    timeline.endDate as string,
+    timeline.endTime as string
+  );
+
+  const patchBody = {
+    //body change accordingly
+    name: test.name, //cannot be changed
+    platform: test.platform,
+    start_time: start_time,
+    end_time: end_time,
+    is_active: "",
+    purpose: {
+      type: "attendance",
+      params: "quiz",
+    },
+    repeat_schedule: { type: "weekly", params: [1, 2, 3, 4, 5, 6, 7] }, // this is hardcoded needs to be changed
+    session_id: "",
+
+    // form_schema_id: session.form_schema_id, // should be signup_form_name
+    signup_form_name: "Haryana Registration Form",
+    type: "sign-in",
+    auth_type: session.auth_type,
+    activate_signup: session.activate_signup,
+    id_generation: session.id_generation,
+    redirection: session.redirection,
+    pop_up_form: session.pop_up_form,
+    number_of_fields_in_pop_form: session.number_of_fields_in_pop_form,
+    meta_data: {
+      group: student.program,
+      batch: student.batch,
+      grade: student.grade,
+      course: student.course,
+      stream: student.stream,
+      test_format: test.format,
+      test_purpose: test.purpose,
+      enabled: timeline.isEnabled,
+      infinite_session: true,
+      cms_test_id:
+        // "https://cms.peerlearning.com/chapter_tests/655df9a23562d97a6300b53e",
+        test.cmsId,
+      test_takers_count: student.testTakers,
+      has_synced_to_bq: false,
+      optional_limits: test.optionalLimit,
+      marking_scheme: test.markingScheme,
+      test_type: test.type,
+      date_created: new Date().toISOString().split("T")[0],
+      admin_testing_link: "",
+    },
+  };
+
+  try {
+    // const response = await instance.patch<PatchTypes>(
+    //   `${process.env.AF_DB_URL}/api/session/${id}`,
+    //   requestBody
+    // );
+
+    // const sessionId = response.data.id;
+    const message = {
+      action: "patch",
+      id: sessionId,
+      patch_session: patchBody,
+    };
+    publishMessage(message);
+    return {
+      id: sessionId,
+    };
+  } catch (error) {
+    console.error("Error posting form data", error);
+    throw error;
+  }
+}
+
+export default async function handler(
+  req: NextApiRequest,
+  res: NextApiResponse
+) {
+  if (req.method === "PATCH") {
+    try {
+      const response = await UpdateFormDataToBackend(req.body);
+      res.status(200).json(response);
+    } catch (error) {
+      console.error("Error in API route", error);
+      res.status(500).json({ error: "Internal Server Error" });
+    }
+  } else {
+    res.status(405).json({ error: "Method Not Allowed" });
+  }
+}
+ */
+
+async function formatDateTime(date: string, time: string) {
```

Comment:
this code is already there in PostFormData ? 

maybe you can have a "single" file for both PostFormData and UpdateFormData since there's so much similarity. avoid unnecessary code repetition

Reply:
merged into a single file